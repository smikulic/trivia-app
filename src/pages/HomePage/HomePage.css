.HomePage {
  text-align: center;
}

.HomePage-logo {
  height: 40vmin;
}

.HomePage-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.HomePage-link {
  color: #09d3ac;
}
