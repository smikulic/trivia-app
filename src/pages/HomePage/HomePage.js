import React from 'react';
import './HomePage.css';

function HomePage({ handleOnStart, loading }) {
  return (
    <div className="HomePage">
      { loading && (
        <header className="HomePage-header">
          Loading the quiz...
        </header>
      )}
      { !loading && (
        <header className="HomePage-header">
          <h1>
            Welcome to the Trivia Challenge!
          </h1>
          <p>
            You will be presented with 10 True or False questions.
          </p>
          <p>
            Can you score 100%?
          </p>
          <div
            className="HomePage-link"
            onClick={() => handleOnStart()}
          >
            Begin
          </div>
        </header>
      )}
    </div>
  );
}

export default HomePage;
