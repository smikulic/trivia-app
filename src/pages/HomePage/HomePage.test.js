import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HomePage from './HomePage';

it('renders without crashing - no props', () => {
  const div = document.createElement('div');
  ReactDOM.render(<HomePage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing - all props', () => {
  const div = document.createElement('div');
  ReactDOM.render(<HomePage handleOnStart={() => undefined} loading={false} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly - loading state', () => {
  const tree = renderer
    .create(<HomePage handleOnStart={() => undefined} loading={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly - loaded state', () => {
  const tree = renderer
    .create(<HomePage handleOnStart={() => undefined} loading={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
