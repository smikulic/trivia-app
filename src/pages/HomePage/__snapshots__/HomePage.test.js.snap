// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - loaded state 1`] = `
Array [
  <h1>
    Welcome to the Trivia Challenge!
  </h1>,
  <p>
    You will be presented with 10 True or False questions.
  </p>,
  <p>
    Can you score 100%?
  </p>,
  <div
    className="app-link"
    onClick={[Function]}
  >
    Begin
  </div>,
]
`;

exports[`renders correctly - loading state 1`] = `"Loading the quiz..."`;
